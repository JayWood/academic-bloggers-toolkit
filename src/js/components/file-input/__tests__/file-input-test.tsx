import * as React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import FileInput, { Props } from '..';

type TestProps = Pick<Props, 'disabled' | 'fill' | 'large' | 'text'>;

const setup = (props?: TestProps) => {
    const onChange = jest.fn();
    const component = shallow(
        <FileInput {...props} inputProps={{ onChange }} />,
    );
    return {
        component,
        onChange,
    };
};

describe('<FileInput />', () => {
    const BASE_COMPONENT = toJSON(setup().component);
    it('should match snapshots', () => {
        expect(BASE_COMPONENT).toMatchSnapshot();

        const { component: disabledVariant } = setup({ disabled: true });
        expect(toJSON(disabledVariant)).toMatchDiffSnapshot(BASE_COMPONENT);

        const { component: fillVariant } = setup({ fill: true });
        expect(toJSON(fillVariant)).toMatchDiffSnapshot(BASE_COMPONENT);

        const { component: largeVariant } = setup({ large: true });
        expect(toJSON(largeVariant)).toMatchDiffSnapshot(BASE_COMPONENT);

        const { component: textVariant } = setup({ text: 'Hello World' });
        expect(toJSON(textVariant)).toMatchDiffSnapshot(BASE_COMPONENT);
    });
    it('should handle change', () => {
        const { component, onChange } = setup();
        let input = component.find('input').first();
        input.simulate('change', 'foo');
        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenCalledWith('foo');

        const noInputChangeHandler = shallow(<FileInput inputProps={{}} />);
        input = noInputChangeHandler.find('input').first();
        expect(() => input.simulate('change', 'foo')).not.toThrowError();
    });
});
