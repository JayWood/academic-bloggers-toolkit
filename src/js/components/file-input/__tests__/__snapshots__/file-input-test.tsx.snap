// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileInput /> should match snapshots 1`] = `
<label
  className="fileInput"
>
  <input
    onChange={[Function]}
    type="file"
  />
  <span
    className="fileUploadInput"
  >
    Choose file...
  </span>
</label>
`;

exports[`<FileInput /> should match snapshots 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,10 +1,9 @@
  <label
-   className=\\"fileInput disabled\\"
+   className=\\"fileInput\\"
  >
    <input
-     disabled={true}
      onChange={[Function anonymous]}
      type=\\"file\\"
    />
    <span
      className=\\"fileUploadInput\\""
`;

exports[`<FileInput /> should match snapshots 3`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,7 +1,7 @@
  <label
-   className=\\"fileInput fill\\"
+   className=\\"fileInput\\"
  >
    <input
      onChange={[Function anonymous]}
      type=\\"file\\"
    />"
`;

exports[`<FileInput /> should match snapshots 4`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,7 +1,7 @@
  <label
-   className=\\"fileInput large\\"
+   className=\\"fileInput\\"
  >
    <input
      onChange={[Function anonymous]}
      type=\\"file\\"
    />"
`;

exports[`<FileInput /> should match snapshots 5`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,8 +6,8 @@
      type=\\"file\\"
    />
    <span
      className=\\"fileUploadInput\\"
    >
-     Hello World
+     Choose file...
    </span>
  </label>"
`;
